flink-java/src/main/java/org/apache/flink/api/java/DataSet.java

```diff
@@ -1336,11 +1335,17 @@ public long count() throws Exception {
 	/**
 	 * Writes a DataSet to the standard output stream (stdout).<br/>
 	 * For each element of the DataSet the result of {@link Object#toString()} is written.
-	 * 
-	 *  @return The DataSink that writes the DataSet.
+	 * This triggers execute() automatically.
 	 */
-	public DataSink<T> print() {
-		return output(new PrintingOutputFormat<T>(false));
+	public void print() throws Exception{
+		try {
```

Comment:
How about removing this try/catch block?

We are getting a very deep nesting of exceptions already. Not sure this wrapping exception adds any information. Why not let the "ProgramInvokationFailed" exception from the execution propagate?